### PR TITLE
tests: Gotest template for keeper tests

### PR DESCRIPTION
**Pull request by AlpinYukseloglu**
Closes: #1444 

## What is the purpose of the change

This PR does two things:
1. Sets a standard for how we intend new table-driven tests to be made
2. Implements this standard into a custom `gotest` template so that contributors can save time on testing boilerplate by using the `Go: Generate Unit Tests For Function` that is built into vscode's Go plugin

The current implementation generates the following test 100% automatically:

Screen Shot 2022-07-09 at 1 12 39 AM

The main work left to do is to find a way to filter out `sdk.Context` and `sdk.AccAddress` params from the generated test structs.
##
How to use this template:
1. Add the template folder in this PR to your .vscode folder
2. Add the following to `settings.json` in your .vscode folder:
```
"go.generateTestsFlags": [
    "-template_dir",
    "<absolute path to `templates` folder>"
  ],
```
An example path to insert above: `"/Users/alpinyukseloglu/GitHub/osmosis/.vscode/templates"`

cc: @ValarDragon – do you think this belongs in the main repo, or do you think it would be more appropriate to keep it entirely contained in our IDE setup docs?

## Brief Changelog

- Add gotest template files to .vscode

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (no)
  - How is the feature or change documented? (not documented)

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -3,6 +3,10 @@
   "go.lintTool": "golangci-lint",
   "go.formatTool": "goimports",
   "go.useLanguageServer": true,
+  "go.generateTestsFlags": [
+    "-template_dir",
+    "/Users/alpinyukseloglu1/Documents/GitHub/osmosis/.vscode/templates"
```
[the remark this answers is not in the document]
Comment (mattverse) [reply .vscode/settings.json:8]: cref: https://github.com/microsoft/vscode/issues/2809

**Comment by ValarDragon** [comment]
@AlpinYukseloglu can we do remaining steps to make this happen?

**Comment by AlpinYukseloglu** [comment]
The only open question was how to handle path instructions for the template – I figured just adding setup instructions to CONTRIBUTING.md would be the simplest approach!

File: .vscode/settings.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (ValarDragon) [reply .vscode/settings.json]: oof yeah this seems to be the case. Thats unfortunate :(

**Comment by ValarDragon** [comment]
SGTM! Definitely crazy this isn't supported in vscode natively O_O

**Review by ValarDragon — APPROVED** [review]

**Pull request merged**